WordPress/gutenberg
# [RNMobile] Simplify media insertion flow - Part 3 component integration

`gutenberg-mobile` PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/2700

- [Part 1](https://github.com/WordPress/gutenberg/pull/29546)
- [Part 2](https://github.com/WordPress/gutenberg/pull/29547)
- [Part 3](https://github.com/WordPress/gutenberg/pull/29548) ( Current ) 

Builds on top of https://github.com/WordPress/gutenberg/pull/29547 - Do not merge until this PR has been merged and the base branch is updated to `trunk`. 

## Description
This change simplifies the media insertion flow by showing the media options sheet when you tap `Image`, `Video` or `Gallery` to add the media placeholder to the canvas. Before you would have to add the media placeholder and click `Add Image / Video / Media` for it to be shown after insertion. 


## Testing 

1. Tap (+) (inserter) – this shows the Block Library sheet
2. Tap Image (or Video or Gallery) to add the block's media placeholder to canvas – which triggers the media upload options sheet. 
3. Notice the media upload options auto open.

### Image

Android | iOS 
--------|-------
 <kbd><img src="https://user-images.githubusercontent.com/1509205/109728900-11892700-7b85-11eb-9420-7c5818926068.gif" width="320"></kbd>    |       <kbd><img src="https://user-images.githubusercontent.com/1509205/109728904-1221bd80-7b85-11eb-9340-fc4ca0bb1daf.gif" width="320"></kbd>

### Video

Android | iOS 
--------|-------
  <kbd><img src="https://user-images.githubusercontent.com/1509205/109728905-12ba5400-7b85-11eb-878f-bfef2ac954df.gif" width="320"></kbd>    |     <kbd><img src="https://user-images.githubusercontent.com/1509205/109728911-1352ea80-7b85-11eb-8d71-856715b27bc4.gif" width="320"></kbd>
  
### Gallery

Android | iOS 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/109728897-1057fa00-7b85-11eb-9649-f02f6beeef19.gif" width="320"></kbd>    |      <kbd><img src="https://user-images.githubusercontent.com/1509205/109728898-10f09080-7b85-11eb-906e-0a2521cc6232.gif" width="320"></kbd>
 
----------

### Regression Testing

1. Verify that undo/redo on all media blocks does not trigger the auto-opening of the media options sheet.
2. Verify that selecting the media-related blocks in the canvas does not trigger the auto opening. ( Note : When you select a media-related block on the emulator, the placeholder may receive the touch event and open the media options sheet.)

## Checklist:
